src/appcontroller.cpp

```diff
@@ -164,4 +166,44 @@ int ApplicationController::UserRunInstallerCallback(const wchar_t* filePath)
     return ms_cbUserRunInstaller(filePath);
 }
 
+int ApplicationController::AlternateAppcastCallback(bool manual, struct Appcast& appcast)
+{
+    // Get the alternate appcast data (use the user defined callback) if one is defined
+    if (!ms_cbAlternateAppcast)
+        return 0;
+
+    // Use temporary storage for receiving the data (to be populated by the callback)
+    static const int appcastBufferLength = 500;
```

Comment:
How is the buffer length calculated? Perhaps add a comment to indicate how we came up with 500.
Is this 500 limit enforced on the server side as well?

Reply:
These are arbitrary limits that I came up with to store those strings as they come across the DLL<->App boundary. They could be anything, they just need to cover our maximum string lengths for those fields. The server has nothing to do with the 500 limit. The other side of the callback in the Windows Client (https://github.com/CovenantEyes/windows-client/pull/708/files#diff-0c693420fa4d706cb0e226144e7133c0842abe54bbbfbeb0997974e0e609a933R1926-R1990) is making sure that the buffer is not overrun using string copy and then checking for any truncation.

I will add some comments to better document the situation.